test/TestSrt.cpp

```diff
@@ -0,0 +1,231 @@
+    ASSERT_TRUE(client.startClient("127.0.0.1", 8009, 16, 1000, 100, ctx, 1456, kPsk));
+
+    // check for client connecting
+    {
+        std::unique_lock<std::mutex> lock(connectedMutex);
+        bool successfulWait = connectedCondition.wait_for(lock, std::chrono::seconds(2), [&]() { return connected; });
+        EXPECT_TRUE(successfulWait) << "Timeout waiting for client to connect";
+    }
+
+    auto nClients = 0;
+    server.getActiveClients([&](std::map<SRTSOCKET, std::shared_ptr<SRTNet::NetworkConnection>>& activeClients) {
+        nClients = activeClients.size();
+    });
+    EXPECT_EQ(nClients, 1);
+
+    // notice when client disconnects
+    std::condition_variable disconnectCondition;
+    std::mutex disconnectMutex;
+    bool disconnected = false;
+    server.clientDisconnected = [&](std::shared_ptr<SRTNet::NetworkConnection>& ctx, SRTSOCKET lSocket) {
+        {
+            std::lock_guard<std::mutex> lock(disconnectMutex);
+            disconnected = true;
+        }
+        disconnectCondition.notify_one();
+    };
+
+    EXPECT_TRUE(client.stop());
+    {
+        std::unique_lock<std::mutex> lock(disconnectMutex);
+        bool successfulWait =
+            disconnectCondition.wait_for(lock, std::chrono::seconds(2), [&]() { return disconnected; });
+        EXPECT_TRUE(successfulWait) << "Timeout waiting for client disconnect";
+    }
+
+    // start a new client and stop the server
+    SRTNet client2;
+    ASSERT_TRUE(client2.startClient("127.0.0.1", 8009, 16, 1000, 100, ctx, 1456, kPsk));
+    disconnected = false;
+    EXPECT_TRUE(server.stop());
+    {
+        std::unique_lock<std::mutex> lock(disconnectMutex);
+        bool successfulWait =
+            disconnectCondition.wait_for(lock, std::chrono::seconds(2), [&]() { return disconnected; });
+        EXPECT_TRUE(successfulWait) << "Timeout waiting for client disconnect";
+    }
+}
+
+TEST(TestSrt, SendReceive) {
+    class ConnectionCtx {};
+
+    SRTNet server;
+    SRTNet client;
+    auto serverCtx = std::make_shared<SRTNet::NetworkConnection>();
+    serverCtx->mObject = std::make_shared<ConnectionCtx>();
+    auto clientCtx = std::make_shared<SRTNet::NetworkConnection>();
+    clientCtx->mObject = std::make_shared<ConnectionCtx>();
+
+    server.clientConnected = [&](struct sockaddr& sin, SRTSOCKET newSocket,
+                                 std::shared_ptr<SRTNet::NetworkConnection>& ctx) {
+        EXPECT_EQ(ctx, serverCtx);
+        return ctx;
+    };
+
+    server.clientDisconnected = [&](std::shared_ptr<SRTNet::NetworkConnection>& ctx, SRTSOCKET lSocket) {
+        EXPECT_EQ(ctx, serverCtx);
+    };
+
+    // start server and client
+    ASSERT_TRUE(server.startServer("127.0.0.1", 8009, 16, 1000, 100, 1456, kPsk, serverCtx));
+    ASSERT_TRUE(client.startClient("127.0.0.1", 8009, 16, 1000, 100, clientCtx, 1456, kPsk));
+
+    std::vector<uint8_t> sendBuffer(1000);
```

Comment:
SRT has a limited message size that is limited by the MTU and the SRT header size. Maybe it is a good idea to add a unit test that checks that sending fails when exceeding this limit. I think the limit is between 1450-1500 bytes.
I also think there is some way you can send messages that exceeds this limit by splitting them into multiple chunks and set the chunk sequence number in msgCtrl structure. I haven't tested it myself so I don't know how (or if) it works. Maybe add a test case for this, as it would be useful for https://jira.edgeware.tv/browse/STAR-140

Reply:
> SRT has a limited message size that is limited by the MTU and the SRT header size. Maybe it is a good idea to add a unit test that checks that sending fails when exceeding this limit. I think the limit is between 1450-1500 bytes. I also think there is some way you can send messages that exceeds this limit by splitting them into multiple chunks and set the chunk sequence number in msgCtrl structure. I haven't tested it myself so I don't know how (or if) it works. Maybe add a test case for this, as it would be useful for https://jira.edgeware.tv/browse/STAR-140

Good point, I will add a test for exceeding the limit.
But I suggest doing the chunking will be for the future. It could be an extension of the wrapper I think and tested when this has been done.

Reply:
Yes, you are right, that is probably too much work for this PR.